### DISCUSSION TITLE
Having Trouble Syncing

### DISCUSSION THREAD

**Discussion by avatar-kaleb** [community/community#10601] [Copilot Conversations] [answered]
Hello,

I was accepted into the copilot beta today, and haven't had any luck getting the synthesizing to produce any solutions for me.

I've tried uninstalling and reinstalling, deleting it directly in `~/vscode`, and even trying out the nightly extension to make sure everything was done propery.

Here's the output:

```
[INFO] [solutions] [2022-01-26T23:05:06.225Z] prompt: {"prefix":"// Language: javascript\n// Path: test.js\nfunction calculateDaysBetweenDates(begin, end) {\n","suffix":""}
[INFO] [default] [2022-01-26T23:05:06.226Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-01-26T23:05:15.210Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:15.237Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:15.278Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 9051 ms
[INFO] [default] [2022-01-26T23:05:15.278Z] prompt: {"prefix":"// Language: javascript\n// Path: test.js\nfunction calculateDaysBetweenDates(begin, end) {\n","suffix":""}
[INFO] [streamChoices] [2022-01-26T23:05:15.281Z] solution 4 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[ERROR] [streamChoices] [2022-01-26T23:05:15.282Z] Error parsing JSON stream data,{"error":{"message":"request cancelled"}}
[INFO] [streamChoices] [2022-01-26T23:05:15.283Z] solution 0 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.283Z] solution 1 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.283Z] solution 2 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.284Z] solution 3 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.284Z] solution 5 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.285Z] solution 6 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.285Z] solution 7 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.285Z] solution 8 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [streamChoices] [2022-01-26T23:05:15.286Z] solution 9 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldaXUlOyX7Iut6IwjHU9IoGkW5G}] created: [{1643238306}]
[INFO] [default] [2022-01-26T23:05:15.773Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:15.816Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:15.944Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:15.982Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:16.114Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:16.165Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:16.289Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:16.818Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:16.948Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:19.623Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:19.765Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-01-26T23:05:19.876Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:20.004Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-01-26T23:05:20.123Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 102 ms
[INFO] [default] [2022-01-26T23:05:20.123Z] prompt: {"prefix":"// Language: javascript\n// Path: test.js\nfunction calculateDaysBetweenDates(begin, end) {","suffix":""}
[INFO] [streamChoices] [2022-01-26T23:05:20.125Z] solution 0 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-4UldoxzKcwKluXQqu2pJh79s8VOBs}] created: [{1643238320}]
[INFO] [ghostText] [2022-01-26T23:05:20.126Z] Breaking, no choices
[INFO] [default] [2022-01-26T23:05:21.089Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-01-26T23:05:31.144Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 10054 ms
[INFO] [default] [2022-01-26T23:05:31.145Z] prompt: {"prefix":"// Language: javascript\n// Path: test.js\nfunction calculateDaysBetweenDates(begin, end) {\n","suffix":""}
[INFO] [streamChoices] [2022-01-26T23:05:31.147Z] solution 1 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-4UldpXUUTwlGzFQITMGmNLwkxZQmi}] created: [{1643238321}]
[INFO] [streamChoices] [2022-01-26T23:05:31.148Z] solution 0 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-4UldpXUUTwlGzFQITMGmNLwkxZQmi}] created: [{1643238321}]
[ERROR] [streamChoices] [2022-01-26T23:05:31.148Z] Error parsing JSON stream data,{"error":{"message":"request timeout"}}
[INFO] [streamChoices] [2022-01-26T23:05:31.148Z] solution 2 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-4UldpXUUTwlGzFQITMGmNLwkxZQmi}] created: [{1643238321}]
[INFO] [ghostText] [2022-01-26T23:05:31.149Z] Breaking, no choices
```

Please let me know if there is anything I can do to get it working

**Comment by vdere** [comment]
Same happens with me. It worked fine until a few days ago. The latency of showing the solutions has also increased. 
Tried signing in and out of github account. 
Reinstalling the copilot extension. 
Nothing works. It is always stuck at "Synthesizing 0/10 solutions". Shows `"Error parsing JSON stream data,{"error":{"message":"request timeout"}}"`. Tried on different machine and network as well but the issue persists. Its not a firewall, network issue as suggested in some threads.

**Reply by jiekeshi** [reply]
Hi, I have the same issues. Have you got any solutions for that?

**Comment by AcmeGamers** [comment, marked as answer]
# Introduction to Problem
Anyone from studying days, this is the best name I could think of lol. Though I was having the same issue (working and not working sometimes) while working on test files in JS, and from the logs that I'm getting, I believe it is due to a timeout error and fetch issues from the GitHub codepiolet database. 

Anyone who could be facing this issue is likely happening because of the timeout error from the server. Based on the report, there are three conclusions I'm getting at:

- It is causing issues while parsing JSON data, and the error is being caused due to a request timeout when making a request to the server.
![Screenshot_921](https://user-images.githubusercontent.com/52010365/151488978-f28e53af-0895-429d-be22-c4c561b0e9ec.png)

- It is able to load properly as long as the suggestion is available locally (in your system).
![Screenshot_919](https://user-images.githubusercontent.com/52010365/151488550-6611d12c-ebf4-4714-923f-2cd183d55be8.png)

- The request is made to the server and the response is obtained (which is when the suggestion is shown to you).
![Screenshot_922](https://user-images.githubusercontent.com/52010365/151489446-064defb7-8016-4b4c-ad5f-71a71dc4e5cb.png)

## So what can be the possible solution?
Now I tried to fix this and checked for what could be the possible fix for this (since it was working fine, why cause issues now lol). So I came up with the following:

- Use a VPN, (I actually wanted to try this, but my problem got fixed before it).

The solution that worked for me:
- Press **Ctrl + Shift + W**, it will close the entire window
- Reopen the window you closed by pressing **Ctrl + Shift + N**, it will reload the extensions. 
- Make a new file in JS and write, `function process()`. 
- When you write `function process()`, let the load complete first
![image](https://user-images.githubusercontent.com/52010365/151492332-348e87b7-8afc-4238-b699-7d7487e33c14.png)
![image](https://user-images.githubusercontent.com/52010365/151492245-ed19ec0b-1054-4f5b-ae4f-8a89f162ffb5.png)

- - After 10 tries of writing the same line `function process()`, again and again, it worked for me fluently and isn't causing issues anymore.

## Conclusion
For people who might be confused about what was happening, I hope this small guide can help you to understand the problem and fix it as well.

**Reply by avatar-kaleb** [reply]
This worked thank you!

**Reply by AcmeGamers** [reply]
You're welcome, I'm glad it worked for ya 🙂

**Reply by Quexint** [reply]
Deactivating and re-activating copilot also works for me. 🙂

**Reply by pritom007** [reply]
`Press Ctrl + Shift + W, it will close the entire window`

`Reopen the window you closed by pressing Ctrl + Shift + N, it will reload the extensions.`


These two just worked like magic. Thanks :D
